spec/support/reservation_helper.rb

```diff
@@ -0,0 +1,23 @@
+require 'spec_helper'
+
+module ReservationHelper
+  def mock_reservation(factory: nil, **attrs)
+    attrs = FactoryGirl.attributes_for(factory).merge attrs if factory
+    instance_spy('reservation', **attrs)
+  end
+
+  def stub_collection_methods(res, *scopes)
+    allow(Reservation).to receive(:all).and_return(res)
+    scopes.each { |s| allow(Reservation).to receive(s).and_return(res) }
+  end
+
+  # NOT to be used with the above method
+  # skips every query but the last one, which will return the given array of res
+  def stub_scope_chain(res, *chain)
```

Comment:
If there's a way to avoid it that's probably best, although in small doses its not the end of the world.